integration/networking/bridge_test.go

```diff
@@ -0,0 +1,331 @@
+package networking
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/integration/internal/container"
+	"github.com/docker/docker/integration/internal/network"
+	"github.com/docker/docker/testutil/daemon"
+	"gotest.tools/v3/assert"
+	"gotest.tools/v3/skip"
+)
+
+// TestBridgeICC tries to ping container ctr1 from container ctr2 using its hostname. Thus, this test checks:
+// 1. DNS resolution ; 2. ARP/NDP ; 3. whether containers can communicate with each other.
+func TestBridgeICC(t *testing.T) {
+	skip.If(t, testEnv.DaemonInfo.OSType == "windows")
+
+	d := daemon.New(t)
+	d.StartWithBusybox(t, "-D", "--experimental", "--ip6tables")
+	defer d.Stop(t)
+
+	c := d.NewClientT(t)
+	defer c.Close()
+
+	testcases := []struct {
+		name       string
+		bridgeOpts []func(create *types.NetworkCreate)
```

Comment:
Nit: arguments don't need to be named in type declarations for functions.
```suggestion
		bridgeOpts []func(*types.NetworkCreate)
```